client/Invoice.js

```diff
@@ -1,37 +1,46 @@
-Template.Invoices.onCreated(function() {
-  this.incrementInvoicesNumber = 30;
-  this.initInvoicesNumber = 30;
-  this.elementsLimit = new ReactiveVar();
-  this.elementsLimit.set(this.initInvoicesNumber);
-   var self = this;
-   self.autorun(function() {
-      self.subscribe('invoices',
-         FlowRouter.getParam("timeRange"),
-         FlowRouter.getQueryParam('sortedBy'),
-         FlowRouter.getQueryParam('sortOrder'),
-         self.elementsLimit.get()
-         );
-   });
-});
-
-Template.Invoices.helpers({
-  elementsLimit : function() {
-    return Template.instance().elementsLimit.get();
+TemplateController('invoices_template', {
+  state: {
+    elementsLimit: 30
+  },
+  private: {
+    incrementInvoicesNumber: 30
+  },
+  onCreated() {
+    this.autorun(() => {
+      this.subscribe('invoices',
+        invoiceQueryState.toJSON(),
+        this.state.elementsLimit()
+      );
+    });
   },
-   invoices: () => {
+  helpers: {
+    invoices: () => {
       return Invoices.byTimeRange(
-         FlowRouter.getParam("timeRange"),
-         FlowRouter.getQueryParam('sortedBy'),
-         FlowRouter.getQueryParam('sortOrder'),
-         this.Template.instance().elementsLimit.get()
-      )},
-   hasMoreContent: () => {
-     return ReactiveMethod.call('hasMoreInvoicesThan', this.Template.instance().elementsLimit.get(),FlowRouter.getParam("timeRange"));
-  }
-});
-
-Template.Invoices.events({
+        invoiceQueryState.toJSON(),
+        this.Template.instance().state.elementsLimit()
+       )},
+    hasMoreContent: () => {
+      return ReactiveMethod.call('hasMoreInvoicesThan', this.Template.instance().state.elementsLimit());
+    },
+    searchFields: () => {
+      return [
+                {"name":"email","value":"","format":"string"},
+                {"name":"invoiceNumber","value":"","format":"integer"},
+                {"name":"total","value":"","format":"integer"}
```

Comment:
You don't have to use `"` to define these fields! just write `{ name: "email", value: "", format: "string" },`